python/Ganga/GPIDev/Lib/File/File.py

```diff
@@ -64,21 +64,11 @@ def __init__(self, name=None, subdir=os.curdir):
         if not subdir is None:
             self.subdir = subdir
 
-    def __construct__(self, args):
-        if len(args) == 1 and isinstance(args[0], str):
-            v = args[0]
-            import os.path
-            expanded = expandfilename(v)
-            # if it is not already an absolute filename
-            if not urlprefix.match(expanded):
-                if os.path.exists(os.path.abspath(expanded)):
-                    self.name = os.path.abspath(expanded)
-                else:
-                    self.name = v
-            else:  # bugfix #20545
-                self.name = expanded
-        else:
-            super(File, self).__construct__(args)
+    def __setattr__(self, attr, value):
+        actual_value = value
+        if attr == "name":
+            actual_value = expandfilename(value)
+        super(File, self).__setattr__(attr, value)
```

Comment:
There's a bug that I think line 71 should read actual_value rather than value (for automatic filename expansion). I think this just passes through in it's current form without doing anything which seems surprising but behind the scenes in Ganga we try to use `File` with absolute filenames and avoid exposing it to the GPI due to now having LocalFile.